packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -205,38 +147,33 @@ html<TableColumnMappingHeaderView>`${x => x.getMappingToRender().groupHeaderView
 
 ```HTML
 <${this.icon}
-    title="${x => x.label}"
-    aria-label="${x => x.label}"
+    title="${x => x.labelHidden ? x.label: null}"
+    aria-label="${x  => x.labelHidden ? x.label: null}"
+    aria-hidden="${x => x.labelHidden ? 'false' : 'true'}"
     severity="${x => x.severity}">
 </${this.icon}>
+${when(() => !x.labelHidden, html`
+    <span>${x => x.label}</span>
+`)}
 ```
 
 `mapping.groupHeaderViewTemplate`:
 
 ```HTML
 <${this.icon}
-    title="${x => x.label}"
-    aria-label="${x => x.label}"
+    aria-hidden="${x => x.labelHidden ? 'false' : 'true'}"
     severity="${x => x.severity}">
 </${this.icon}>
-<span
-    ${ref('span')}
-    @mouseover="${x => {
-        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
-    }}"
-    @mouseout="${x => {
-        x.isValidContentAndHasOverflow = false;
-    }}"
-    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+<span>
     ${x => x.label}
 </span>`;
 ```
 
 ### Grouping
 
-Grouping will be based on the record value. The grouping header will display the rendered icon/spinner/text. In the case of an icon/spinner, it will also be followed by the `label` text. This will disambiguate cases where multiple record values map to the same icon (assuming the labels are different).
+Grouping will be based on the record value. The grouping header will display the rendered label along with the icon/spinner, if configured.
 
-For values that do not match any mapping, we will display the raw data value. While this introduces inconsistency, it seems preferable to the alternative, which is having multiple, separate groupings with a blank header (well, with just the item count in parens). Even in the case where there is a default mapping, we would still end up with separate groups with the identical default mapped icon and/or text, which is just as bad.
+For values that do not match any mapping, the group row will display only the count. This is considered invalid data from the table's perspective and should be fixed within the client application.
```

Comment:
Note: This is not a proposed behavior change. I'm simply updating the HLD to reflect the decisions we made while initially implementing the column.